src/Symfony/Component/DependencyInjection/Compiler/RegisterFactoryPass.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Attribute\Autoconfigure;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Exception\AutoconfigureFailedException;
+use Symfony\Component\DependencyInjection\Loader\YamlFileLoader;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\ExpressionLanguage\Expression;
+
+/**
+ * Register factories for services tagged with "container.factory".
+ *
+ * @author Maelan Le Borgne <maelan.leborgne@gmail.com>
+ */
+final class RegisterFactoryPass implements CompilerPassInterface
+{
+    private static \Closure $registerForAutoconfiguration;
+
+    public function process(ContainerBuilder $container): void
+    {
+        foreach ($container->findTaggedServiceIds('container.factory') as $id => $attributes) {
+            $definition = $container->getDefinition($id);
+            if (!$definition->hasTag('container.ignore_attributes')) {
```

Comment:
Remove and check if this is done automaticaly in the registerForAutoconfiguration